lib/src/nested_form_builder.dart

```diff
@@ -0,0 +1,133 @@
+
+  /// The widget below this widget in the tree.
+  ///
+  /// This is the root of the widget hierarchy that contains this form.
+  ///
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  /// Used to enable/disable form fields auto validation and update their error
+  /// text.
+  ///
+  /// {@macro flutter.widgets.form.autovalidateMode}
+  final AutovalidateMode? autovalidateMode;
+
+  /// An optional Map of field initialValues. Keys correspond to the field's
+  /// name and value to the initialValue of the field.
+  ///
+  /// The initialValues set here will be ignored if the field has a local
+  /// initialValue set.
+  final Map<String, dynamic>? initialValue;
+
+  /// Whether the form should ignore submitting values from fields where
+  /// `enabled` is `false`.
+  /// This behavior is common in HTML forms where _readonly_ values are not
+  /// submitted when the form is submitted.
+  ///
+  /// When `true`, the final form value will not contain disabled fields.
+  /// Default is `false`.
+  final bool skipDisabled;
+
+  /// Whether the form is able to receive user input.
+  ///
+  /// Defaults to true.
+  ///
+  /// When `false` all the form fields will be disabled - won't accept input -
+  /// and their enabled state will be ignored.
+  final bool enabled;
+
+  /// Whether the form should auto focus on the first field that fails validation.
+  final bool autoFocusOnValidationFailure;
+
+  /// Whether to clear the internal value of a field when it is unregistered.
+  ///
+  /// Defaults to `false`.
+  ///
+  /// When set to `true`, the form builder will not keep the internal values
+  /// from disposed [FormBuilderField]s. This is useful for dynamic forms where
+  /// fields are registered and unregistered due to state change.
+  ///
+  /// This setting will have no effect when registering a field with the same
+  /// name as the unregistered one.
+  final bool clearValueOnUnregister;
+
+  NestedFormBuilder({
+    Key? key,
+    required this.name,
+    required this.child,
+    this.onChanged,
+    this.onWillPop,
+    this.autovalidateMode,
+    this.initialValue,
+    this.skipDisabled = false,
+    this.enabled = true,
+    this.autoFocusOnValidationFailure = false,
+    this.clearValueOnUnregister = false,
+    GlobalKey<FormBuilderState>? formKey,
+    this.parentFormKey,
+    this.valueTransformer,
+  })  : formKey = formKey ?? GlobalKey<FormBuilderState>(),
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) => FormBuilderField<Map<String, dynamic>>(
+        name: name,
+        initialValue:
+            parentFormKey?.currentState?.initialValue[name] ?? initialValue,
+        valueTransformer: valueTransformer ?? (_) => _,
+        onReset: () => formKey.currentState?.reset(),
+        builder: (field) => FormBuilder(
+          key: formKey,
+          initialValue: field.value ?? {},
+          onChanged: () {
+            final st = formKey.currentState;
```

Comment:
What means `st`? I think that is better use more autoexplaines names or delete the variable

Reply:
just for debug, `formKey.currentState` is too long